lib/ecto/changeset.ex

```diff
@@ -436,7 +436,9 @@ defmodule Ecto.Changeset do
   During casting, all `permitted` parameters whose values match the specified
   type information will have their key name converted to an atom and stored
   together with the value as a change in the `:changes` field of the changeset.
-  All parameters that are not explicitly permitted are ignored.
+  If the cast value matches the default value for the schema, it will not be
+  included in the `:changes` unless the `:force_changes: true` option is
+  provided. All parameters that are not explicitly permitted are ignored.
```

Comment:
```suggestion
  If the cast value matches the current value for the field, it will not be
  included in the `:changes` unless the `:force_changes: true` option is
  provided. All parameters that are not explicitly permitted are ignored.
```

Sorry one last nitpick. The change won't be registered if it doesn't alter the current data, which in the case of a new struct is the default value.